### PR TITLE
Rename AggregateRoot to Aggregate

### PR DESCRIPTION
**Pull request by Corniel**
It is shorter, and as even DDD experts can not fully agree on the distinction, it is better this way.

### CODE REVIEW

**Comment by CptWesley** [comment]
You could consider keeping an `AggregateRoot` abstract class that extends `Aggregate` with a deprecation notice, so that you don't instantly kill all existing implementations.

**Comment by Corniel** [comment]
> You could consider keeping an `AggregateRoot` abstract class that extends `Aggregate` with a deprecation notice, so that you don't instantly kill all existing implementations.

You only have to rename, that should not be too hard, should it?

**Comment by CptWesley** [comment]
> > You could consider keeping an `AggregateRoot` abstract class that extends `Aggregate` with a deprecation notice, so that you don't instantly kill all existing implementations.
> 
> You only have to rename, that should not be too hard, should it?

I understand, but it might discourage people to upgrade

**Comment by Corniel** [comment]
> I understand, but it might discourage people to upgrade

Fair point to. I changed the behavior.

File: README.md [hunk 1]
```diff
@@ -20,43 +20,45 @@ current state of an aggregate root can always be achieved by replaying these
 events. 
 
 ## Always Valid
-Aggregate roots should always be valid according to the boundaries of their
-domain. There are multiple ways to achieve this, but within Qowaiv Domain Model
-this is guaranteed via an implicitly triggered validator.
+The aggregate should always be valid according to the boundaries of their domain.
```
Comment (LauraXiulan) [inline README.md:23]: Aggregate is singular so I would say 'its domain'

File: README.md [hunk 2]
```diff
@@ -67,15 +69,15 @@ trick. This might be beneficial for aggregate roots that have a lot of events
 (think thousands, or more).
 
 ## Event dispatcher
-The aggregate root has an event dispatcher that by default uses compiled
+Each aggregate has an event dispatcher that by default uses compiled
 expressions to execute matching non-public `When(@event)` methods. This is
 extremely fast, but if desired, by implementing a own `EventDispatcher`, the
```
Comment (LauraXiulan) [inline README.md:74]: ... your own `EventDispatcher`, ...

File: README.md [hunk 3]
```diff
@@ -67,15 +69,15 @@ trick. This might be beneficial for aggregate roots that have a lot of events
...
 
 ## Example 1
-A (simplified) real life example of a financial entry, using `AggregateRoot<TAggregate, TId>`:
+A (simplified) real life example of a financial entry, using `Aggregate<TAggregate, TId>`:
 ``` C#
-public sealed class FinancialEntry : AggregateRoot<FinancialEntry, Guid>
+public sealed class FinancialEntry : Aggregate<FinancialEntry, Guid>
 {
     // Note that FinancialEntryValidator is passed in for validation purposes.
     // See the implementation of this class below.
```
Comment (LauraXiulan) [inline README.md:83]: I can't put a comment there, but there is a typo in line 158/160 "`Result<TAggregate>` is up to the developer"

File: src/Qowaiv.DomainModel.TestTools/AggregateRootAssert.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@ namespace Qowaiv.DomainModel.TestTools;
 /// <summary>Assertions on the aggregate root.</summary>
 public static class AggregateRootAssert
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel.TestTools/AggregateRootAssert.cs:8]: AggregateAssert (+ summary)

File: src/Qowaiv.DomainModel.TestTools/AggregateRootAssert.cs [hunk 5]
```diff
@@ -22,7 +22,7 @@ public static class AggregateRootAssert
     /// </param>
     [Assertion]
     public static void HasUncommittedEvents<TAggregate, TId>(Result<TAggregate> actualAggregate, params object[] expectedEvents)
-        where TAggregate : AggregateRoot<TAggregate, TId>, new()
+        where TAggregate : Aggregate<TAggregate, TId>, new()
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel.TestTools/AggregateRootAssert.cs:25]: The sb.AppendLine below still mentions aggregate root.

File: src/Qowaiv.DomainModel/Aggregate.cs [hunk 6]
```diff
@@ -1,9 +1,9 @@
 ﻿namespace Qowaiv.DomainModel;
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel/Aggregate.cs:1]: Weird characters here

File: src/Qowaiv.DomainModel/Aggregate_TAggregate.cs [hunk 7]
```diff
@@ -28,7 +28,7 @@ protected AggregateRoot(IValidator<TAggregate> validator)
     protected static ImmutableCollection Events => ImmutableCollection.Empty;
 #pragma warning restore S2743 // Static fields should not be used in generic types
 
-    /// <summary>The dynamic </summary>
+    /// <summary>The dispatcher that calls the methods that (re)play the different events.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel/Aggregate_TAggregate.cs:31]: Line 83 has a reference to aggregate root

File: src/Qowaiv.DomainModel/Aggregate_TAggregate_TId.cs [hunk 8]
```diff
@@ -7,17 +7,17 @@
 /// <typeparam name="TId">
 /// The type of the identifier.
 /// </typeparam>
-public class AggregateRoot<TAggregate, TId> : AggregateRoot<TAggregate>
-    where TAggregate : AggregateRoot<TAggregate, TId>, new()
+public class Aggregate<TAggregate, TId> : Aggregate<TAggregate>
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel/Aggregate_TAggregate_TId.cs:10]: In the summary it still mentions aggregate root

Comment (LauraXiulan) [reply src/Qowaiv.DomainModel/Aggregate_TAggregate_TId.cs:10]: In line 20 as well

File: src/Qowaiv.DomainModel/Obsolete.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
+﻿namespace Qowaiv.DomainModel;
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel/Obsolete.cs:1]: Weird characters

File: src/Qowaiv.DomainModel.TestTools/AggregateRootAssert.cs (hunk 5 above)
Comment (Corniel) [reply src/Qowaiv.DomainModel.TestTools/AggregateRootAssert.cs:25]: Will be dropped with #37

**Review by CptWesley — APPROVED** [review]

File: src/Qowaiv.DomainModel/Aggregate_TAggregate_TId.cs (hunk 8 above)
Comment (Corniel) [reply src/Qowaiv.DomainModel/Aggregate_TAggregate_TId.cs:10]: done!

**Review by LauraXiulan — APPROVED** [review]

**Pull request merged**